src/handler/update.ts

```diff
@@ -57,9 +56,10 @@ export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayPr
       return addHttpHeaders(vinErrors);
     }
 
+    const archiveNeeded = !((recordFromDB.techRecord_statusCode === StatusCode.CURRENT && requestBody.techRecord_statusCode === StatusCode.PROVISIONAL));
     const [updatedRecordFromDB, updatedNewRecord] = await processUpdateRequest(recordFromDB, requestBody, userDetails);
 
-    const record = await updateVehicle([updatedRecordFromDB] as TechRecordGet[], updatedNewRecord as TechRecordGet);
+    const record = await updateVehicle(archiveNeeded ? [updatedRecordFromDB] as TechRecordGet[] : [], updatedNewRecord as TechRecordGet);
```

Comment:
nitpick:

would be nicer as
```suggestion
    const recordsToArchive = archiveNeeded ? [updatedRecordFromDB] as TechRecordGet[] : []
    const record = await updateVehicle(recordsToArchive, updatedNewRecord as TechRecordGet);
```